## 38. 프로젝트 루트 확인


In [162]:
from pathlib import Path

project_root = Path.cwd()

if project_root.name == "notebooks":
    project_root = project_root.parent

data_dir = project_root / "data" / "raw"

print("프로젝트 루트:", project_root)
print("데이터 폴더:", data_dir)
print("데이터 폴더 존재:", data_dir.exists())

프로젝트 루트: c:\Dev\ai-data-analysis
데이터 폴더: c:\Dev\ai-data-analysis\data\raw
데이터 폴더 존재: True


## 39. 필수 파일 확인

In [163]:
required_files = [
    "customers.csv",
    "products.csv",
    "orders.csv",
    "order_items.csv",
]

for file_name in required_files:
    file_path = data_dir / file_name
    print(file_name, file_path.exists(), file_path.stat().st_size if file_path.exists() else 0)

customers.csv True 5641
products.csv True 4103
orders.csv True 11693
order_items.csv True 15356


## 40. 데이터 불러오기

In [164]:

import pandas as pd

customers = pd.read_csv(data_dir / "customers.csv")
products = pd.read_csv(data_dir / "products.csv")
orders = pd.read_csv(data_dir / "orders.csv")
order_items = pd.read_csv(data_dir / "order_items.csv")

## 41. DataFrame 사전 만들기

In [165]:

datasets = {
    "customers": customers,
    "products": products,
    "orders": orders,
    "order_items": order_items,
}
# 딕셔너리라는 타입으로 정리 하는 작업
# print(datasets)

## 42. 크기 요약

In [166]:
shape_summary = pd.DataFrame(
    [
        {
            "dataset": name,
            "rows": df.shape[0], 
            "columns": df.shape[1],
        }
        for name, df in datasets.items()
    ]

)

print(shape_summary)
# py파일에는 그냥 shape_summary만 쓰면 에러남 그래서 꼭 print()로 출력해야함

       dataset  rows  columns
0    customers   150        6
1     products   100        4
2       orders   301        5
3  order_items   765        5


## 43. 한 행의 의미 기록
- custmers: 고객 한 명
- products: 상품 한 개
- orders: 주문 한 건
- order_items: 주문에 포함된 상품 한 항목

## 44. 처음과 마지막 행 확인

In [167]:

for name, df in datasets.items():
    print(f"\n[{name}] 앞부분")
    display(df.head())
    print(f"[{name}] 뒷부분")
    display(df.tail())


[customers] 앞부분


,customer_id,name,gender,age,city,signup_date
0,1,김수민,F,19,광주,2024-07-16
1,2,김정호,F,32,대구,2025-11-29
2,3,이경수,F,61,성남,2024-07-09
3,4,조영호,F,55,울산,2026-05-10
4,5,이예원,F,19,부산,2024-10-10


[customers] 뒷부분


,customer_id,name,gender,age,city,signup_date
145,146,김숙자,M,61,성남,2026-01-19
146,147,이정남,M,19,부산,2025-03-10
147,148,오도현,M,29,고양,2026-07-12
148,149,김정자,M,20,부산,2024-11-15
149,150,조미영,M,40,대전,2025-12-31



[products] 앞부분


,product_id,product_name,category,price
0,1,전자기기 상품 001,전자기기,160000
1,2,도서 상품 002,도서,34000
2,3,전자기기 상품 003,전자기기,152000
3,4,생활용품 상품 004,생활용품,70000
4,5,식품 상품 005,식품,186000


[products] 뒷부분


,product_id,product_name,category,price
95,96,생활용품 상품 096,생활용품,112000
96,97,전자기기 상품 097,전자기기,154000
97,98,스포츠 상품 098,스포츠,10000
98,99,뷰티 상품 099,뷰티,200000
99,100,도서 상품 100,도서,102000



[orders] 앞부분


,order_id,customer_id,order_date,payment_method,order_status
0,1,123,2026-06-03,card,completed
1,2,77,2025-08-19,naver_pay,cancelled
2,3,138,2025-12-16,bank_transfer,cancelled
3,4,57,2026-02-26,kakao_pay,cancelled
4,5,125,2026-01-17,card,cancelled


[orders] 뒷부분


,order_id,customer_id,order_date,payment_method,order_status
296,297,22,2026-02-11,bank_transfer,completed
297,298,64,2026-01-21,naver_pay,cancelled
298,299,135,2025-09-04,bank_transfer,completed
299,300,96,2026-03-24,kakao_pay,cancelled
300,301,300,2026-03-24,kakao_pay,cancelled



[order_items] 앞부분


,order_item_id,order_id,product_id,quantity,unit_price
0,1,1,100,3,102000
1,2,1,87,5,25000
2,3,1,7,3,142000
3,4,1,9,3,193000
4,5,2,72,4,189000


[order_items] 뒷부분


,order_item_id,order_id,product_id,quantity,unit_price
760,761,298,40,4,174000
761,762,299,8,2,189000
762,763,299,12,4,175000
763,764,300,59,5,32000
764,765,302,200,10,332222


## 45. 컬럼명 확인

In [168]:

for name, df in datasets.items():
    print(name, df.columns.tolist())

customers ['customer_id', 'name', 'gender', 'age', 'city', 'signup_date']
products ['product_id', 'product_name', 'category', 'price']
orders ['order_id', 'customer_id', 'order_date', 'payment_method', 'order_status']
order_items ['order_item_id', 'order_id', 'product_id', 'quantity', 'unit_price']


## 46. 컬럼명 공백 점검

In [169]:

for name, df in datasets.items():
    suspicious = [
        col for col in df.columns
        if col != col.strip()
    ]

    print(name, suspicious)

customers []
products []
orders []
order_items []


## 47. info() 실행

In [170]:
for name, df in datasets.items():

    print(f"\n===== {name} =====")

    df.info()


===== customers =====
<class 'pandas.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype
---  ------       --------------  -----
 0   customer_id  150 non-null    int64
 1   name         150 non-null    str  
 2   gender       150 non-null    str  
 3   age          150 non-null    int64
 4   city         150 non-null    str  
 5   signup_date  150 non-null    str  
dtypes: int64(2), str(4)
memory usage: 7.2 KB

===== products =====
<class 'pandas.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   product_id    100 non-null    int64
 1   product_name  100 non-null    str  
 2   category      100 non-null    str  
 3   price         100 non-null    int64
dtypes: int64(2), str(2)
memory usage: 3.3 KB

===== orders =====
<class 'pandas.DataFrame'>
RangeIndex: 301 entries, 0 to 300
Data columns (total 5 columns):

## 48. dtype 요약표

In [171]:
dtype_rows = []

for name, df in datasets.items():
    for column, dtype in df.dtypes.items():
        dtype_rows.append(
            {
                "dataset": name,
                "column": column,
                "dtype": str(dtype),
            }
        )
dtype_summary = pd.DataFrame(dtype_rows)
dtype_summary

,dataset,column,dtype
0,customers,customer_id,int64
1,customers,name,str
2,customers,gender,str
3,customers,age,int64
4,customers,city,str
5,customers,signup_date,str
6,products,product_id,int64
7,products,product_name,str
8,products,category,str
9,products,price,int64


## 49. 날짜형 후보 확인

In [172]:
date_columns = {
    "customers": ["signup_date"],
    "orders": ["order_date"],
}
for dataset_name, columns in date_columns.items():
    df = datasets[dataset_name]
    for column in columns:
        converted = pd.to_datetime(
            df[column],
            errors="coerce"
        )
        print(
            dataset_name,
            column,
            "원본 dtype:", df[column].dtype,
            "변환 실패:", converted.isna().sum()

        )

customers signup_date 원본 dtype: str 변환 실패: 0
orders order_date 원본 dtype: str 변환 실패: 0


## 50. 결측치 개수

In [173]:
for name, df in datasets.items():

    print(f"\n[{name}]")

    display(df.isna().sum().to_frame("missing_count"))


[customers]


,missing_count
customer_id,0
name,0
gender,0
age,0
city,0
signup_date,0



[products]


,missing_count
product_id,0
product_name,0
category,0
price,0



[orders]


,missing_count
order_id,0
customer_id,0
order_date,0
payment_method,0
order_status,0



[order_items]


,missing_count
order_item_id,0
order_id,0
product_id,0
quantity,0
unit_price,0


## 51. 결측치 비율

In [174]:
missing_summaries = []
for name, df in datasets.items():
    summary = pd.DataFrame(
        {
            "missing_count": df.isna().sum(),
            "missing_rate_pct": (
                df.isna().mean() * 100
            ).round(2),
        }
    )

    summary.insert(0, "dataset", name)
    summary.insert(1, "column", summary.index)
    missing_summaries.append(summary.reset_index(drop=True))

missing_summary = pd.concat(
    missing_summaries,
    ignore_index=True

)

print(missing_summary)

        dataset          column  missing_count  missing_rate_pct
0     customers     customer_id              0               0.0
1     customers            name              0               0.0
2     customers          gender              0               0.0
3     customers             age              0               0.0
4     customers            city              0               0.0
5     customers     signup_date              0               0.0
6      products      product_id              0               0.0
7      products    product_name              0               0.0
8      products        category              0               0.0
9      products           price              0               0.0
10       orders        order_id              0               0.0
11       orders     customer_id              0               0.0
12       orders      order_date              0               0.0
13       orders  payment_method              0               0.0
14       orders    order_

## 52. 전체 행 중복

In [175]:
for name, df in datasets.items():
    print(name, "전체 중복:", df.duplicated().sum())

customers 전체 중복: 0
products 전체 중복: 0
orders 전체 중복: 0
order_items 전체 중복: 0


## 53. 주요 ID 점검

In [176]:
primary_keys = {
    "customers": "customer_id",
    "products": "product_id",
    "orders": "order_id",
    "order_items": "order_item_id",
}

id_check_rows = []

for name, key in primary_keys.items():
    df = datasets[name]
    id_check_rows.append(
        {
            "dataset": name,
            "key": key,
            "missing": df[key].isna().sum(),
            "duplicates": df[key].duplicated().sum(),
            "unique": df[key].nunique(),
            "rows": len(df),
        }

    )

id_check_summary = pd.DataFrame(id_check_rows)

id_check_summary

,dataset,key,missing,duplicates,unique,rows
0,customers,customer_id,0,0,150,150
1,products,product_id,0,0,100,100
2,orders,order_id,0,0,301,301
3,order_items,order_item_id,0,0,765,765


## 54. 발견 내용을 기록합니다

## 데이터 품질 발견 기록

| 데이터셋 | 컬럼 | 문제 종류 | 건수 | 해석 | 처리 여부 |

|---|---|---|---:|---|---|

| | | | | | Chapter 05에서 결정 |


| 데이터셋        | 컬럼          | 문제 종류    |      건수 | 해석                           | 처리 여부           |
| ----------- | ----------- | -------- | ------: | ---------------------------- | --------------- |
| customers   | signup_date | 날짜 변환 실패 | 0| 가입날짜 이상없음|N/A|
| orders      | order_date  | 날짜 변환 실패 | 0| 주문날짜 이상없음|N/A
| customers   | (해당 컬럼)     | 결측치      | 0| 결측치 없음        | N/A
| products    | (해당 컬럼)     | 결측치      | 0| 결측치 없음       | N/A
| orders      | 전체 행        | 중복 데이터   | 0|  중복행 없음             | N/A
| customers   | customer_id | PK 중복    | 0| 중복키 없음             | N/A
| orders      | customer_id | 참조 무결성   | 0| customers에 없는 customer_id 참조 | N/A
| order_items | product_id  | 참조 무결성   | 0| products에 없는 product_id 참조   | N/A


## 55. 숫자형 기본 통계

In [177]:
print(customers[["age"]].describe())
print(products[["price"]].describe())
print(order_items[["quantity", "unit_price"]].describe())

              age
count  150.000000
mean    42.086667
std     15.613166
min     19.000000
25%     29.000000
50%     40.000000
75%     57.000000
max     69.000000
               price
count     100.000000
mean   110040.000000
std     56433.910574
min      5000.000000
25%     65750.000000
50%    112000.000000
75%    161000.000000
max    200000.000000
         quantity     unit_price
count  765.000000     765.000000
mean     3.062745  108853.884967
std      1.432142   57530.605659
min      1.000000    5000.000000
25%      2.000000   62000.000000
50%      3.000000  111000.000000
75%      4.000000  162000.000000
max     10.000000  332222.000000


## 56. 범주형 값 확인

In [178]:
category_columns = {
    "customers": ["gender", "city"],
    "products": ["category"],
    "orders": ["payment_method", "order_status"],
}

for dataset_name, columns in category_columns.items():
    df = datasets[dataset_name]
    for column in columns:
        print(f"\n[{dataset_name}.{column}]")
        display(
            df[column]
            .value_counts(dropna=False)
            .to_frame("count")
        )


[customers.gender]


,count
gender,
F,84
M,66



[customers.city]


,count
city,
성남,21
광주,17
부산,16
대구,15
서울,15
울산,14
인천,14
대전,14
수원,13



[products.category]


,count
category,
스포츠,19
전자기기,17
생활용품,16
뷰티,16
도서,14
패션,11
식품,7



[orders.payment_method]


,count
payment_method,
kakao_pay,80
naver_pay,77
bank_transfer,74
card,70



[orders.order_status]


,count
order_status,
completed,184
cancelled,65
refunded,52


## 57. 날짜 범위 확인

In [179]:
signup_dates = pd.to_datetime(
    customers["signup_date"],
    errors="coerce"
)

order_dates = pd.to_datetime(
    orders["order_date"],
    errors="coerce"
)

print("가입일 변환 실패:", signup_dates.isna().sum())
print("가입일 범위:", signup_dates.min(), "~", signup_dates.max())
print("주문일 변환 실패:", order_dates.isna().sum())
print("주문일 범위:", order_dates.min(), "~", order_dates.max())

가입일 변환 실패: 0
가입일 범위: 2023-08-10 00:00:00 ~ 2026-07-24 00:00:00
주문일 변환 실패: 0
주문일 범위: 2025-08-05 00:00:00 ~ 2026-08-04 00:00:00


## 58. 날짜 관계 점검 준비

주문일과 고객 가입일을 직접 비교하려면 두 데이터를 연결해야 합니다. 

이번 장에서는 구조와 키가 연결 가능한지 확인하고, 실제 병합과 분석은 이후 장에서 본격적으로 진행합니다


## 59. 참조하지 못한 고객ID

In [180]:
unknown_customer_ids = sorted(
    set(orders["customer_id"].dropna())
    - set(customers["customer_id"].dropna())
)

print("존재하지 않는 customer_id:", unknown_customer_ids)

존재하지 않는 customer_id: [300]


## 60. 참조하지 못한 주문 ID

In [181]:
unknown_order_ids = sorted(
    set(order_items["order_id"].dropna())
    - set(orders["order_id"].dropna())
)

print("존재하지 않는 order_id:", unknown_order_ids)

존재하지 않는 order_id: [302]


## 61. 참조하지 못한 상품 ID

In [182]:
unknown_product_ids = sorted(
    set(order_items["product_id"].dropna())
    - set(products["product_id"].dropna())
)

print("존재하지 않는 product_id:", unknown_product_ids)

존재하지 않는 product_id: [200]


## 62. 반복 점검 함수

In [183]:
def inspect_dataframe(
    name: str,
    df: pd.DataFrame,
    primary_key: str | None = None,
) -> dict:
    result = {
        "dataset": name,
        "rows": df.shape[0],
        "columns": df.shape[1],
        "column_names": df.columns.tolist(),
        "dtypes": {
            column: str(dtype)
            for column, dtype in df.dtypes.items()
        },
        "missing_total": int(df.isna().sum().sum()),
        "duplicate_rows": int(df.duplicated().sum()),
    }

    if primary_key is not None:
        result["primary_key"] = primary_key
        result["key_missing"] = int(
            df[primary_key].isna().sum()
        )

        result["key_duplicates"] = int(
            df[primary_key].duplicated().sum()
        )

        result["key_unique"] = int(
            df[primary_key].nunique()
        )

    return result

In [184]:
inspection_results = []
for name, df in datasets.items():
    inspection_results.append(
        inspect_dataframe(
            name,
            df,
            primary_keys[name],
        )
    )
inspection_results

[{'dataset': 'customers',
  'rows': 150,
  'columns': 6,
  'column_names': ['customer_id',
   'name',
   'gender',
   'age',
   'city',
   'signup_date'],
  'dtypes': {'customer_id': 'int64',
   'name': 'str',
   'gender': 'str',
   'age': 'int64',
   'city': 'str',
   'signup_date': 'str'},
  'missing_total': 0,
  'duplicate_rows': 0,
  'primary_key': 'customer_id',
  'key_missing': 0,
  'key_duplicates': 0,
  'key_unique': 150},
 {'dataset': 'products',
  'rows': 100,
  'columns': 4,
  'column_names': ['product_id', 'product_name', 'category', 'price'],
  'dtypes': {'product_id': 'int64',
   'product_name': 'str',
   'category': 'str',
   'price': 'int64'},
  'missing_total': 0,
  'duplicate_rows': 0,
  'primary_key': 'product_id',
  'key_missing': 0,
  'key_duplicates': 0,
  'key_unique': 100},
 {'dataset': 'orders',
  'rows': 301,
  'columns': 5,
  'column_names': ['order_id',
   'customer_id',
   'order_date',
   'payment_method',
   'order_status'],
  'dtypes': {'order_id': 'int6

## 63. LLM에 전달할 안전한 구조 요약

In [185]:
safe_schema_summary = {
    "datasets": inspection_results,
    "relationships": {
        "orders.customer_id": {
            "references": "customers.customer_id",
            "unknown_count": len(unknown_customer_ids),
        },
        "order_items.order_id": {
            "references": "orders.order_id",
            "unknown_count": len(unknown_order_ids),
        },

        "order_items.product_id": {
            "references": "products.product_id",
            "unknown_count": len(unknown_product_ids),
        },
    },
}

print(safe_schema_summary)

{'datasets': [{'dataset': 'customers', 'rows': 150, 'columns': 6, 'column_names': ['customer_id', 'name', 'gender', 'age', 'city', 'signup_date'], 'dtypes': {'customer_id': 'int64', 'name': 'str', 'gender': 'str', 'age': 'int64', 'city': 'str', 'signup_date': 'str'}, 'missing_total': 0, 'duplicate_rows': 0, 'primary_key': 'customer_id', 'key_missing': 0, 'key_duplicates': 0, 'key_unique': 150}, {'dataset': 'products', 'rows': 100, 'columns': 4, 'column_names': ['product_id', 'product_name', 'category', 'price'], 'dtypes': {'product_id': 'int64', 'product_name': 'str', 'category': 'str', 'price': 'int64'}, 'missing_total': 0, 'duplicate_rows': 0, 'primary_key': 'product_id', 'key_missing': 0, 'key_duplicates': 0, 'key_unique': 100}, {'dataset': 'orders', 'rows': 301, 'columns': 5, 'column_names': ['order_id', 'customer_id', 'order_date', 'payment_method', 'order_status'], 'dtypes': {'order_id': 'int64', 'customer_id': 'int64', 'order_date': 'str', 'payment_method': 'str', 'order_status'

## 64. LLM 구조 점검 프롬프트


나는 온라인 쇼핑몰 데이터 분석을 준비하고 있습니다.
아래 정보는 원본 행이 아니라 구조 점검 요약입니다.

{'datasets': [{'dataset': 'customers', 'rows': 150, 'columns': 6, 'column_names': ['customer_id', 'name', 'gender', 'age', 'city', 'signup_date'], 'dtypes': {'customer_id': 'int64', 'name': 'str', 'gender': 'str', 'age': 'int64', 'city': 'str', 'signup_date': 'str'}, 'missing_total': 0, 'duplicate_rows': 0, 'primary_key': 'customer_id', 'key_missing': 0, 'key_duplicates': 0, 'key_unique': 150}, {'dataset': 'products', 'rows': 100, 'columns': 4, 'column_names': ['product_id', 'product_name', 'category', 'price'], 'dtypes': {'product_id': 'int64', 'product_name': 'str', 'category': 'str', 'price': 'int64'}, 'missing_total': 0, 'duplicate_rows': 0, 'primary_key': 'product_id', 'key_missing': 0, 'key_duplicates': 0, 'key_unique': 100}, {'dataset': 'orders', 'rows': 301, 'columns': 5, 'column_names': ['order_id', 'customer_id', 'order_date', 'payment_method', 'order_status'], 'dtypes': {'order_id': 'int64', 'customer_id': 'int64', 'order_date': 'str', 'payment_method': 'str', 'order_status': 'str'}, 'missing_total': 0, 'duplicate_rows': 0, 'primary_key': 'order_id', 'key_missing': 0, 'key_duplicates': 0, 'key_unique': 301}, {'dataset': 'order_items', 'rows': 765, 'columns': 5, 'column_names': ['order_item_id', 'order_id', 'product_id', 'quantity', 'unit_price'], 'dtypes': {'order_item_id': 'int64', 'order_id': 'int64', 'product_id': 'int64', 'quantity': 'int64', 'unit_price': 'int64'}, 'missing_total': 0, 'duplicate_rows': 0, 'primary_key': 'order_item_id', 'key_missing': 0, 'key_duplicates': 0, 'key_unique': 765}], 'relationships': {'orders.customer_id': {'references': 'customers.customer_id', 'unknown_count': 1}, 'order_items.order_id': {'references': 'orders.order_id', 'unknown_count': 1}, 'order_items.product_id': {'references': 'products.product_id', 'unknown_count': 1}}}

print(safe_schema_summary)
다음을 검토해 주세요.

1. 각 데이터셋에서 한 행이 무엇을 의미할 가능성이 높은지 설명해 주세요.
2. 주요 ID와 외래 키를 구분해 주세요.
3. 즉시 삭제하면 안 되는 결측치·중복 사례를 알려 주세요.
4. 추가로 점검할 구조 문제를 우선순위대로 제안해 주세요.
5. 실제 데이터에 없는 컬럼이나 관계를 가정하지 마세요.
6. 각 제안에 대해 확인할 pandas 코드의 목적만 설명해 주세요.

개인정보나 원본 고객 행은 제공하지 않았습니다.

## 수정전

| 검증 항목                  | 결과 |
| ---------------------- | -: |
| customers 기본키 중복       | 0건 |
| products 기본키 중복        | 0건 |
| orders 기본키 중복          | 0건 |
| order_items 기본키 중복     | 0건 |
| 전체 결측치                 | 0건 |
| 전체 행 중복                | 0건 |
| 존재하지 않는 customer_id 참조 | 1건 |
| 존재하지 않는 order_id 참조    | 1건 |
| 존재하지 않는 product_id 참조  | 1건 |


## 수정 후
| 구분      | 점검 항목                                            | 결과            | 해석               | 권장 조치                 |
| ------- | ------------------------------------------------ | ------------- | ---------------- | --------------------- |
| 데이터 구조  | customers                                        | 고객 1명의 정보     | 고객 기준 테이블        | 유지                    |
| 데이터 구조  | products                                         | 상품 1개의 정보     | 상품 기준 테이블        | 유지                    |
| 데이터 구조  | orders                                           | 주문 1건         | 주문 기준 테이블        | 유지                    |
| 데이터 구조  | order_items                                      | 주문에 포함된 상품 1개 | 주문 상세 테이블        | 유지                    |
| 기본키(PK) | customer_id, product_id, order_id, order_item_id | 모두 중복 및 결측 없음 | 기본키 무결성 양호       | 유지                    |
| 외래키(FK) | orders.customer_id → customers.customer_id       | 관계 확인         | 고객-주문 연결         | 유지                    |
| 외래키(FK) | order_items.order_id → orders.order_id           | 관계 확인         | 주문-주문상세 연결       | 유지                    |
| 외래키(FK) | order_items.product_id → products.product_id     | 관계 확인         | 상품-주문상세 연결       | 유지                    |
| 결측치     | 전체 데이터셋                                          | 0건            | 결측치 없음           | 추가 조치 불필요             |
| 중복      | 전체 데이터셋                                          | 0건            | 중복 데이터 없음        | 추가 조치 불필요             |
| 참조 무결성  | orders.customer_id                               | 1건            | 존재하지 않는 고객 ID 참조 | 원인 확인 후 처리            |
| 참조 무결성  | order_items.order_id                             | 1건            | 존재하지 않는 주문 ID 참조 | 원인 확인 후 처리            |
| 참조 무결성  | order_items.product_id                           | 1건            | 존재하지 않는 상품 ID 참조 | 원인 확인 후 처리            |
| 추가 점검   | 날짜 컬럼                                            | 점검 권장         | 날짜 변환 및 범위 확인    | `pd.to_datetime()` 활용 |
| 추가 점검   | 자료형(dtype)                                       | 점검 완료         | 자료형 적절성 확인       | `info()` 활용           |
| 추가 점검   | 범주형 데이터                                          | 점검 권장         | 오타 및 예상 외 값 확인   | `value_counts()` 활용   |
| 추가 점검   | 숫자형 데이터                                          | 점검 권장         | 이상치 및 분포 확인      | `describe()` 활용       |
